src/components/AlertModal/AlertModal.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { FunctionComponent, useMemo } from "react";
+import { Modal, StyleSheet, Text, View } from "react-native";
+import AlertLogo from "../../../assets/icons/alert.svg";
+import { DarkButton } from "../Layout/Buttons/DarkButton";
+import { SecondaryButton } from "../Layout/Buttons/SecondaryButton";
+import { DangerButton } from "../Layout/Buttons/DangerButton";
+import { size } from "../../common/styles";
+
+const styles = StyleSheet.create({
+  centeredView: {
+    flex: 1,
+    width: "100%",
+    justifyContent: "center",
+    alignItems: "center",
+    alignSelf: "center",
+    backgroundColor: "rgba(0,0,0,0.6)"
+  },
+  modalView: {
+    marginBottom: 0,
+    paddingTop: 24,
+    width: 280,
+    backgroundColor: "white",
+    borderRadius: size(1),
+    shadowColor: "#000",
+    shadowOffset: {
+      width: 0,
+      height: 2
+    },
+    shadowOpacity: 0.25,
+    shadowRadius: 3.84,
+    elevation: 5
+  },
+  alertIcon: {
+    flexDirection: "column",
+    justifyContent: "center",
+    alignItems: "center",
+    alignSelf: "center",
+    marginBottom: 9
+  },
+  modalTitle: {
+    fontWeight: "bold",
+    fontFamily: "brand-bold",
+    fontSize: 20,
+    color: "#305367",
+    textAlign: "center",
+    marginHorizontal: 28,
+    marginBottom: 4
+  },
+  modalText: {
+    marginHorizontal: 32.5,
+    marginBottom: 32,
+    textAlign: "center",
+    fontSize: 16,
+    color: "#305367"
+  },
+  modalGroupButton: {
+    flexDirection: "row",
+    marginHorizontal: 16,
+    marginBottom: 24
+  },
+  modalSecondaryBtm: {
+    marginRight: 8,
+    flexShrink: 1
+  },
+  modalPrimaryButton: {
+    flex: 1
+  }
+});
+type AlertType = "ERROR" | "WARN" | "CONFIRM" | "INFO";
+
+// Add on the collection of action button texts
+// try to declare object as { [key:string]: CallToActionButtonTexts }
+// but somehow CallToActionKey unable to determine the keys on compile time
+const callToActionCollection = Object.freeze({
+  YES_NO: { primaryActionText: "YES", secondaryActionText: "NO" },
+  OK_CANCEL: { primaryActionText: "OK", secondaryActionText: "CANCEL" },
+  CONFIRM_CANCEL: {
+    primaryActionText: "CONFIRM",
+    secondaryActionText: "CANCEL"
+  }
+});
+
+interface CallToActionButtonTexts {
+  primaryActionText: string;
+  secondaryActionText?: string;
+}
+
+export type CallToActionKey = keyof typeof callToActionCollection;
+
+export interface AlertModalProp {
+  alertType: AlertType;
+  title: string;
+  description: string;
+  buttonTextType: CallToActionKey;
+  visible: boolean;
+  onOk: () => void;
+  onCancel?: () => void;
+  onExit?: any;
+}
+
+export const AlertModal: FunctionComponent<AlertModalProp> = (
+  props: AlertModalProp
+) => {
+  const callToAction = useMemo(
+    () =>
+      callToActionCollection[props.buttonTextType] as CallToActionButtonTexts,
+    [props.buttonTextType]
+  );
```

Comment:
I noticed a few usages of `useMemo`, could you share more about why it's needed here?

Reply:
similar concept as `useCallback`. But now i think again whether its necessary to "memorize" function or value for such a transient component. What do you think?

Reply:
Yea there are overheads with using `useCallback` and `useMemo`, perhaps we can defer the use of this and measure first? These components are afterall relatively small and aren't computationally expensive, and they don't trigger many other re-renders.

I had another comment about the contexts using `useCallback`s, I defaulted to ensuring context values are wrapped properly because a re-render in a context triggers many re-renders all the way down the tree. Though I haven't benchmarked or measured the perf yet, I've encountered some issues with context re-rendering too many times

Reply:
Initially, the reason I used the `useMemo` here, is to cater for users who stubbornly keep trying to execute the same action and facing the same error. But I realised that by clearing the alert modal, the `props` got overridden anyway, kinda defeat the purpose of the `useMemo`. So I will be removing the `useMemo` here. 
Based on the above [comments](https://github.com/rationally-app/mobile-application/pull/128#discussion_r465454367) , lets do a proper perf monitoring and implement `memorized` as needed.